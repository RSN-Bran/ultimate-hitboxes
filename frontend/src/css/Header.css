/*General Header settings*/
#header {
	top: 0%;
	width: 100%;
}

h3 {
	margin-top: 0;
	margin-bottom: .5em;
}

h5 {
	margin-top: 0;
	margin-bottom: 0;
}

#exitInfo {
	position: absolute;
	left: 0%;
	top: 0%;
	cursor: pointer;
}

#help, #links {
	width: 20%;
	display: inline-block;
}

#title {
	width: 60%;
	display: inline-block;
}

.helpButtons {
	cursor: pointer;
	float: left;
}

.linkButtons {
	cursor: pointer;
	float: right;
}

#settingsButton {
	border-radius: 50%;
	-webkit-transition: -webkit-transform .8s ease-in-out;

}

#settingsButton:hover {
	-webkit-transform: rotate(360deg);
	transform: rotate(360deg);
}

@media (min-width: 500px) {
	#exitInfo {
		width: 3%
	}

	.helpButtons, .linkButtons {
		width: 15%
	}

	#twitter, #settingsButton {
		margin-left: 5%;
	}
}

/*Sizing for Mobile*/
@media (max-width: 500px) {
	#exitInfo {
		width: 7%
	}

	.helpButtons, .linkButtons {
		width: 40%;
	}

	#twitter, #settingsButton {
		margin-left: 10%;
	}

	h3 {
		font-size: 15px;
	}
}