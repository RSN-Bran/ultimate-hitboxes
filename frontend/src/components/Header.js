//Import React Elements
import * as React from "react"
import { useHistory } from "react-router-dom";
import {Link, Router, Route, BrowserRouter, Switch} from 'react-router-dom'

//Import CSS
import '../css/Header.css';

//Import Media
import IMAGES from '../media/media_imports.js'

function Header(props) {
  let history = useHistory();

  return (

    //Main Header Object contains three portions from left to right
    //help - Contains Info and Settings Buttons
    //title - Contains site name
    //links - Contains links to github and twitter pages
    
    <div id="header">
      <div id="help">
        <Link to="/info">
          <img id="infoButton"
          className="helpButtons"
          src={IMAGES[`info_${props.settings.theme}`]}/>
        </Link>
        <Link to="/settings">
          <img id="settingsButton"
          className="helpButtons"
          src={IMAGES[`settings_${props.settings.theme}`]}/>
        </Link>
        <img id="backButton"
        className="helpButtons"
        src={IMAGES[`back_${props.settings.theme}`]}
        onClick={() => {history.goBack()}} />

      </div>

      <div id="title">
        <Link to="/">
          <h3>Smash Ultimate Hitbox Viewer</h3>
        </Link>
      </div>

      <div id="links">
        <a href="https://twitter.com/SSBUHitboxes">
          <img id="twitter"
          className="linkButtons"
          src={IMAGES.twitter}/>
        </a>
        <a href="https://github.com/RSN-Bran/ultimate-hitboxes">
          <img id="github"
          className="linkButtons"
          src={IMAGES[`github_${props.settings.theme}`]}/>
        </a>
        <a href="https://discord.gg/jZ9EKJpwde">
          <img id="discord"
          className="linkButtons"
          src={IMAGES.discord} />
        </a>
      </div>
    </div>
      
  )

}

export default Header