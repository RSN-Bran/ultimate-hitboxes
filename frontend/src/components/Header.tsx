//Import React Elements
import * as React from "react"
import { useHistory } from "react-router-dom";
import { BrowserRouter as Router, Link, Route } from 'react-router-dom'

//Import CSS
import '../css/Header.css';

//Import Media
import IMAGES from '../media/media_imports.js'

//Set Social Media imports
const discord = IMAGES.discord
const twitter = IMAGES.twitter
const github = [IMAGES.github_dark, IMAGES.github_light]

//Set Header Imports
const settings = [IMAGES.settings_dark, IMAGES.settings_light]
const info = [IMAGES.info_dark, IMAGES.info_light]
const back = [IMAGES.back_dark, IMAGES.back_light]

function Header(props) {
  let history = useHistory();

  return (

    //Main Header Object contains three portions from left to right
    //help - Contains Info and Settings Buttons
    //title - Contains site name
    //links - Contains links to github and twitter pages
    <div id="header">
      <div id="help">
        <Link to="/info">
          <img id="infoButton" className="helpButtons" src={info[props.dark_light]}/>
        </Link>

        <Link to="/settings">
          <img id="settingsButton" className="helpButtons" src={settings[props.dark_light]}/>
        </Link>

        <img id="backButton" className="helpButtons" src={back[props.dark_light]} onClick={() => {history.goBack()}} />

      </div>

      <div id="title">
        <Link to="/">
          <h3>Smash Ultimate Hitbox Viewer</h3>
        </Link>
      </div>

      <div id="links">
        <a href="https://twitter.com/SSBUHitboxes">
          <img id="twitter" className="linkButtons" src={twitter}/>
        </a>
        <a href="https://github.com/RSN-Bran/ultimate-hitboxes">
          <img id="github" className="linkButtons" src={github[props.dark_light]}/>
        </a>
        <a href="https://discord.gg/jZ9EKJpwde">
          <img id="discord" className="linkButtons" src={discord} />
        </a>
      </div>
    </div>
      
  )

}

export default Header