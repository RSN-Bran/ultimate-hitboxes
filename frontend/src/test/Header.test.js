import React from 'react'
//import {render} from "@testing-library/react"
import {create, act} from 'react-test-renderer';
import {shallow} from 'enzyme'

import Header from '../components/Header'

import "@testing-library/jest-dom/extend-expect"
import { BrowserRouter  } from 'react-router-dom'

import * as mockCharacterData from './mockData/mock-characterdata.json'
import * as mockMoveData from './mockData/mock-movedata.json'

// describe ('Buttons component', () => {
//     it('Buttons exist', () => {
//         const wrapper = shallow(<Buttons
//              currentCharacterData={mockCharacterData}
//              currentMoveData={mockMoveData}
//              settings={{theme:"dark"}}
//              playing={true}
//              currentFrame={4}
//             />)
//     })

// })

describe('Header Snapshot Tests', () => {
    it('Dark mode snapshot', () => {
        let app;
        act(()=>{
            app = create(<BrowserRouter >
                    <Header
                        settings={{theme:"dark"}}
                    />
               </BrowserRouter >)
        })
        expect(app.toJSON()).toMatchSnapshot()
    })

    it('Light mode snapshot', () => {
        let app;
        act(()=>{
            app = create(<BrowserRouter >
                    <Header
                        settings={{theme:"light"}}
                    />
               </BrowserRouter >)
        })
        expect(app.toJSON()).toMatchSnapshot()
    })    
})