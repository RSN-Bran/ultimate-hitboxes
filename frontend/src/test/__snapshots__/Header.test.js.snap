// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Test 1`] = `
<div
  id="header"
>
  <div
    id="help"
  >
    <a
      href="/info"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="infoButton"
      />
    </a>
    <a
      href="/settings"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="settingsButton"
      />
    </a>
    <img
      className="helpButtons"
      id="backButton"
      onClick={[Function]}
    />
  </div>
  <div
    id="title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h3>
        Smash Ultimate Hitbox Viewer
      </h3>
    </a>
  </div>
  <div
    id="links"
  >
    <a
      href="https://twitter.com/SSBUHitboxes"
    >
      <img
        className="linkButtons"
        id="twitter"
      />
    </a>
    <a
      href="https://github.com/RSN-Bran/ultimate-hitboxes"
    >
      <img
        className="linkButtons"
        id="github"
      />
    </a>
    <a
      href="https://discord.gg/jZ9EKJpwde"
    >
      <img
        className="linkButtons"
        id="discord"
      />
    </a>
  </div>
</div>
`;

exports[`Header Snapshot Tests Dark mode snapshot 1`] = `
<div
  id="header"
>
  <div
    id="help"
  >
    <a
      href="/info"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="infoButton"
      />
    </a>
    <a
      href="/settings"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="settingsButton"
      />
    </a>
    <img
      className="helpButtons"
      id="backButton"
      onClick={[Function]}
    />
  </div>
  <div
    id="title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h3>
        Smash Ultimate Hitbox Viewer
      </h3>
    </a>
  </div>
  <div
    id="links"
  >
    <a
      href="https://twitter.com/SSBUHitboxes"
    >
      <img
        className="linkButtons"
        id="twitter"
      />
    </a>
    <a
      href="https://github.com/RSN-Bran/ultimate-hitboxes"
    >
      <img
        className="linkButtons"
        id="github"
      />
    </a>
    <a
      href="https://discord.gg/jZ9EKJpwde"
    >
      <img
        className="linkButtons"
        id="discord"
      />
    </a>
  </div>
</div>
`;

exports[`Header Snapshot Tests Light mode snapshot 1`] = `
<div
  id="header"
>
  <div
    id="help"
  >
    <a
      href="/info"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="infoButton"
      />
    </a>
    <a
      href="/settings"
      onClick={[Function]}
    >
      <img
        className="helpButtons"
        id="settingsButton"
      />
    </a>
    <img
      className="helpButtons"
      id="backButton"
      onClick={[Function]}
    />
  </div>
  <div
    id="title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h3>
        Smash Ultimate Hitbox Viewer
      </h3>
    </a>
  </div>
  <div
    id="links"
  >
    <a
      href="https://twitter.com/SSBUHitboxes"
    >
      <img
        className="linkButtons"
        id="twitter"
      />
    </a>
    <a
      href="https://github.com/RSN-Bran/ultimate-hitboxes"
    >
      <img
        className="linkButtons"
        id="github"
      />
    </a>
    <a
      href="https://discord.gg/jZ9EKJpwde"
    >
      <img
        className="linkButtons"
        id="discord"
      />
    </a>
  </div>
</div>
`;
